schema_based_on_diagram.sql

```diff
@@ -0,0 +1,53 @@
+CREATE TABLE patients (
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    name VARCHAR(255),
+    date_of_birth DATE
+);
+
+CREATE TABLE medical_histories (
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    admitted_at TIMESTAMP,
+    patient_id INT,
+    status VARCHAR(255),
+    FOREIGN KEY (patient_id) REFERENCES patients(id)
+);
+
+CREATE TABLE treatments (
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    type VARCHAR(255),
+    name VARCHAR(255)
+);
+
+CREATE TABLE invoices (
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    total_amount DECIMAL(10,2),
+    generated_at TIMESTAMP,
+    payed_at TIMESTAMP,
+    medical_history_id INT,
+    FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id)
+);
+
+CREATE TABLE invoice_items (
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    UNIT_PRICE DECIMAL(10,2),
+    quantity INT,
+    total_price DECIMAL(10,2),
+    invoice_id INT,
+    treatment_id INT,
+    FOREIGN KEY (invoice_id) REFERENCES invoices(id),
+    FOREIGN KEY (treatment_id) REFERENCES treatments(id)
+);
+
+CREATE TABLE medical_history_treatment (
+    medical_history_id INT,
+    treatment_id INT,
+    PRIMARY KEY (medical_history_id, treatment_id),
+    FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id),
+    FOREIGN KEY (treatment_id) REFERENCES treatments(id)
+);
+
+CREATE INDEX idx_medical_history_id ON invoices(medical_history_id);
+CREATE INDEX idx_patients_name ON patients(name);
+CREATE INDEX idx_treatments_type ON treatments(type);
+CREATE INDEX idx_invoices_generated_at ON invoices(generated_at);
```

Comment:
👏 Good progress on adding tables as per project requirements! 👏 However, there's a point to address regarding indexes. Here's a suggestion for improvement:

- [ ] **Review and Update Indexes:** As per project requirements, indexes are typically created on foreign keys. While the indexes at line numbers 49 and 53 are correct, those at line numbers 50-52 might not align with the project requirements. Review your tables' implementation and create proper indexes based on the foreign keys used.

By ensuring that indexes are appropriately set, you'll enhance the performance and efficiency of your database. Keep up the great work! 🚀🔍